db/migrate/20160405194617_switch_to_medium_text.rb

```diff
@@ -3,23 +3,44 @@
 #
 class SwitchToMediumText < ActiveRecord::Migration[4.2]
   def change
-    # changing batch_uploaded_source_records
-    change_column :batch_uploaded_source_records, :bibtex_source_data, :text, limit: 16_777_215
-    change_column :batch_uploaded_source_records, :error_message, :text, limit: 16_777_215
+    if ActiveRecord::Base.connection.adapter_name.match?(/mysql/i)
+      # changing batch_uploaded_source_records
+      change_column :batch_uploaded_source_records, :bibtex_source_data, :text, limit: 16_777_215
+      change_column :batch_uploaded_source_records, :error_message, :text, limit: 16_777_215
 
-    # changing publications
-    change_column :publications, :xml, :text, limit: 16_777_215
+      # changing publications
+      change_column :publications, :xml, :text, limit: 16_777_215
 
-    # changing pubmed_source_records
-    change_column :pubmed_source_records, :source_data, :text, limit: 16_777_215
+      # changing pubmed_source_records
+      change_column :pubmed_source_records, :source_data, :text, limit: 16_777_215
 
-    # changing sciencewire_source_records
-    change_column :sciencewire_source_records, :source_data, :text, limit: 16_777_215
+      # changing sciencewire_source_records
+      change_column :sciencewire_source_records, :source_data, :text, limit: 16_777_215
 
-    # changing user_submitted_source_records
-    change_column :user_submitted_source_records, :source_data, :text, limit: 16_777_215
+      # changing user_submitted_source_records
+      change_column :user_submitted_source_records, :source_data, :text, limit: 16_777_215
 
-    # changing versions
-    change_column :versions, :object, :text, limit: 16_777_215
+      # changing versions
+      change_column :versions, :object, :text, limit: 16_777_215
+    else
+      # changing batch_uploaded_source_records
+      change_column :batch_uploaded_source_records, :bibtex_source_data, :text
+      change_column :batch_uploaded_source_records, :error_message, :text
```

Comment:
With PG as the DB in all envs, we no longer care about this limit (which was added earlier to allow both MySQL and sqlite to be used).